# SQL Server Rule: SRD0067
  
|    |    |
|----|----|
| Assembly | SqlServer.Rules |
| Namespace | SqlServer.Rules.Design |
| Class | UseCapitalizedKeywordsRule |
  
## Rule Information
  
|    |    |
|----|----|
| Id | SRD0067 |
| Friendly Name | Use capitalized keywords for enhanced readability. |
| Category | Design |
| Ignorable | false |
| Applicable Types | Procedure  |
|   | Scalar Function |
|   | Table |
|   | Table Valued Function |
|   | View |
  
## Description
  
Use capitalized keywords for enhanced readability.
  
## Summary
  
Use capitalized keywords.
  
### Examples
  

Examples of **incorrect** code for this rule:
```tsql
select* from foo;
```
Examples of ** correct** code for this rule:
```tsql
SELECT * FROM foo;
```
  
```sql
create table foo
( 
	ID VARCHAR(64) not null
	, Name VARCHAR(128) NOT NULL
	, Value VARCHAR(256) NOT NULL
);
GO

CREATE TABLE bar
( 
	ID varchar(64)
);
GO

CREATE PROCEDURE hip
AS
SELECT foo FROM foo 
		WHERE foo = 'foo' and
		foo != 'foo'

-- SRD0067
```
```sql
CREATE PROCEDURE dbo.TestSelectStarBeginEndBlock
as
set nocount on;
begin
	SELECT * FROM dbo.TestTableSSDT
end

-- SML005,SRD0067
```
  
### Remarks
  
Capitalizing SQL keywords enhances readability and provides a clear separation between keywords and objects.
  
<sub><sup>Generated by a tool</sup></sub>
