# SQL Server Rule: SRD0067
  
|    |    |
|----|----|
| Assembly | SqlServer.Rules |
| Namespace | SqlServer.Rules.Design |
| Class | UseCapitalizedKeywordsRule |
  
## Rule Information
  
|    |    |
|----|----|
| Id | SRD0067 |
| Friendly Name | Use capitalized keywords for enhanced readability. |
| Category | Design |
| Ignorable | false |
| Applicable Types | Procedure  |
|   | Scalar Function |
|   | Table |
|   | Table Valued Function |
|   | View |
  
## Description
  
Use capitalized keywords for enhanced readability.
  
## Summary
  
Use capitalized keywords.
  
### Examples
  

Examples of **incorrect** code for this rule:
```tsql
select * from foo;
```
Examples of ** correct** code for this rule:
```tsql
SELECT * FROM foo;
```
  
```sql
create table foo
( 
	ID VARCHAR(64) not null
	, Name VARCHAR(128) NOT NULL
	, Valu VARCHAR(256) NOT NULL
);
GO

CREATE TABLE bar
( 
	ID varchar(64)
);
GO

CREATE PROCEDURE hip
AS
SELECT foo FROM foo 
		WHERE foo = 'foo' and
		foo != 'foo'
GO

CREATE TABLE [dbo].[xxx]
(
	[Id] [uniqueidentifier] NOT NULL,
    [CASEID] [varchar](44) NOT NULL,
	[PROCES] [varchar](3) NULL,
    CONSTRAINT [PK_Case] PRIMARY KEY CLUSTERED ([Id] ASC)
)
GO

CREATE VIEW dbo.CaseView
AS
SELECT
     [CASEID] AS [Id]
	, CASE [STATUS]
		WHEN 0 THEN 'X'
	  END AS [Status]
FROM [dbo].[Case]

-- SRD0067
```
```sql
CREATE PROCEDURE dbo.TestSelectStarBeginEndBlock
as
set nocount on;
begin
	SELECT * FROM dbo.TestTableSSDT
end

-- SML005,SRD0067
```
  
### Remarks
  
Capitalizing SQL keywords enhances readability and provides a clear separation between keywords and objects.
  
<sub><sup>Generated by a tool</sup></sub>
