# SQL Server Rule: SRD0069
  
|    |    |
|----|----|
| Assembly | SqlServer.Rules |
| Namespace | SqlServer.Rules.Design |
| Class | AvoidFutureKeywordsRule |
  
## Rule Information
  
|    |    |
|----|----|
| Id | SRD0069 |
| Friendly Name | Avoid the use of future keywords as identifiers. |
| Category | Design |
| Ignorable | false |
| Applicable Types | Procedure  |
|   | Scalar Function |
|   | Table |
|   | Table Valued Function |
|   | View |
  
## Description
  
Avoid the use of future keywords as identifiers.
  
## Summary
  
Avoid future keywords.
  
### Examples
  

Examples of **incorrect** code for this rule:
```tsql
CREATE TABLE bad (absolute int);
```
Examples of ** correct** code for this rule:
```tsql
CREATE TABLE good (absalute int);
```
  
```sql
CREATE TABLE dbo.bad (absolute INT CONSTRAINT [PK_bad] PRIMARY KEY);
GO
CREATE TABLE dbo.good (absalute INT CONSTRAINT [PK_good] PRIMARY KEY);
GO

-- SRD0069
```
  
### Remarks
  
Future keywords could be reserved in future releases of SQL Server as new features are implemented. Consider avoiding the use of these words as identifiers.
  
<sub><sup>Generated by a tool</sup></sub>
