# SQL Server Rule: SRD0066
  
|    |    |
|----|----|
| Assembly | SqlServer.Rules |
| Namespace | SqlServer.Rules.Design |
| Class | UseBeginEndRule |
  
## Rule Information
  
|    |    |
|----|----|
| Id | SRD0066 |
| Friendly Name | BEGIN and END symbols inside conditional statements. |
| Category | Design |
| Ignorable | true |
| Applicable Types | Procedure  |
|   | Scalar Function |
|   | Table Valued Function |
|   | View |
  
## Description
  
Use BEGIN and END symbols inside conditional statements.
  
## Summary
  
Use BEGIN and END symbols inside conditional statements.
  
### Examples
  

Examples of **incorrect** code for this rule:
```tsql
IF(@parm = 1)
SELECT @output = 'foo'
```
Examples of **correct** code for this rule:
```tsql
IF (@parm = 1)
BEGIN
SELECT @output = 'foo'
END
```
  
```sql
CREATE PROCEDURE dbo.BeginEnd
    @parm INT
AS
SET NOCOUNT ON;

IF(@parm = 1)
  SELECT 'foo';
ELSE
  SELECT 'fix';

IF (@parm = 2)
BEGIN
  SELECT 'bar';
END;

-- SRD0066
```
  
### Remarks
  
Use BEGIN and END to bind conditional statments as a single block of code.
  
<sub><sup>Generated by a tool</sup></sub>
